File: app/src/main/java/com/lazycouple/restapiclient/ui/contract/RequestHistoryContract.java
package com.lazycouple.restapiclient.ui.contract;

import com.lazycouple.restapiclient.ui.presenter.BasePresenter;
import com.lazycouple.restapiclient.ui.viewModel.RequestHistoryViewModel;

/**
 * Created by noco on 2016-10-27.
 */
public interface RequestHistoryContract {

    interface View {
        void showList();
        void showRestRequset(String id);
    }

    interface Presenter extends BasePresenter {
        void init();
        void loadList();
        RequestHistoryViewModel getViewModel();
        void onClickedItem(int position);
        void clearItems();
    }

}

File: app/src/main/java/com/lazycouple/restapiclient/ui/contract/RequestHistoryContract.kt
package com.lazycouple.restapiclient.ui.contract

import com.lazycouple.restapiclient.ui.presenter.BasePresenter

/**
 * Created by noco on 2016-10-27.
 */
interface RequestHistoryContract {

    interface View {
        fun showList()
        fun showRestRequset(id: String?)
    }

    interface Presenter : BasePresenter {
        fun init()
        fun loadList()
        fun onClickedItem(position: Int)
        fun clearItems()
        fun destroy()
    }

}
